dsl/src/test/scala/asobu/dsl/ExtractorSpec.scala

```diff
@@ -17,21 +17,36 @@ class ExtractorSpec extends PlaySpecification {
 
     val extractor = fromJson[Foo].body
 
-    val result = extractor(FakeRequest().withJsonBody(Json.obj("bar" → "hello", "bar2" → 3)))
+    val result = extractor.run(FakeRequest().withJsonBody(Json.obj("bar" → "hello", "bar2" → 3))).value
 
     result.map(_.getOrElse(HNil)) must be_==("bar" ->> "hello" :: "bar2" ->> 3 :: HNil).await
 
   }
 
   "and combines two extractor" >> { implicit ee: ExecutionEnv ⇒
+    val extractor1 = fromJson[Foo].body
+
+    val w = Witness("bar3")
+
+    val extractor2 = Extractor.from(bar3 = (_: Request[AnyContent]).headers("bar3"))
+
+    val extractor = extractor1 and extractor2
+
+    val result = extractor.run(FakeRequest().withJsonBody(Json.obj("bar" → "hello", "bar2" → 3)).withHeaders("bar3" → "bar3Value")).value
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Non ascii characters are not allowed](https://www.codacy.com/app/iHeartRadio/asobu/file/2261096929/issues/source?bid=3037259&fileBranchId=3178877#l35)